routers/web/web.go

```diff
@@ -1584,6 +1585,18 @@ func registerRoutes(m *web.Router) {
 			m.Get("/*", context.RepoRefByType(""), repo.Home) // "/*" route is deprecated, and kept for backward compatibility
 		}, repo.SetEditorconfigIfExists)
 
+		// Add a /tree/* path to redirect to the deprecated /src/*
+		// path. This emulates both Github & Gitlab's URL structure.
+		m.Get("/tree/*", func(ctx *context.Context) {
+			prefix := "/"
+			if setting.AppSubURL != "" {
+				prefix = setting.AppSubURL
+			}
```

Comment:
It could be simplified to something like `sprintf("/%s/%s/%s/src/%s", AppSubURL, escape(username), escape(reponame), pathsegmentescape(*))`